rust/src/treefile.rs

```diff
@@ -502,6 +507,25 @@ fn treefile_parse_recurse<P: AsRef<Path>>(
             }
         }
     }
+    if let Some(conditional_includes) = parsed.config.base.conditional_include.take() {
+        for conditional_include in conditional_includes {
+            let matches = match conditional_include.condition {
+                IncludeConditions::Single(c) => c.evaluate(variables)?,
+                IncludeConditions::Multiple(v) => v
+                    .iter()
+                    .map(|c| c.evaluate(variables))
+                    .collect::<Result<Vec<bool>>>()?
+                    .into_iter()
```

Comment:
A `collect()` :arrow_right: `into_iter()` chain creates an unnecessary intermediate allocation.  

How about: `.iter().find(|v| !matches!(v, Ok(true))).transpose()?.is_none()` or so?  

(This may actually be better done as a `fn evaluate_conditionals(variables: BTreeMap)` method in `impl IncludeConditions`)

Reply:
Yeah, the `collect()` was purely for its ability to handle `Result<>`. I think the `matches!` approach would work, but it would ignore errors on conditions which wouldn't change the result, where I think we still want to do semantic checks on all the conditions.

Ahh yes! I just found exactly what we need for this: `try_fold`.